File: software/modules/athena/src/main/java/totah/lab/athena/pocket/compare/AlignmentInitialization.java
package totah.lab.athena.pocket.compare;

/**
 * How a pocket alignment was initialized.
 *
 * <ul>
 *     <li>{@link #PCA_ICP}: principal-axis coarse alignment followed by
 *     ICP refinement (the production default, no sequence
 *     evidence).</li>
 *     <li>{@link #SEQUENCE_SEEDED_KABSCH}: rigid Kabsch fit over
 *     sequence-aligned pocket residue pairs.</li>
 *     <li>{@link #SEQUENCE_SEEDED_KABSCH_ICP}: sequence-seeded Kabsch
 *     followed by ICP refinement that strictly improved the mean
 *     bidirectional distance.</li>
 * </ul>
 */
public enum AlignmentInitialization {
    PCA_ICP,
    SEQUENCE_SEEDED_KABSCH,
    SEQUENCE_SEEDED_KABSCH_ICP
}

File: software/apps/web-ui/src/features/similar/alignmentInitialization.ts
export function alignmentInitializationLabel(initialization: string): string {
  switch (initialization) {
    case 'PCA_ICP':
      return 'PCA + ICP'
    case 'SEQUENCE_SEEDED_KABSCH':
      return 'sequence-seeded Kabsch'
    case 'SEQUENCE_SEEDED_KABSCH_ICP':
      return 'sequence-seeded Kabsch + ICP'
    default:
      return initialization
  }
}
